### PR TITLE
Feature/169999903 add filetype request parameter to the download controller

### PR DESCRIPTION
**Pull request by lingyun1010**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java [hunk 1]
```diff
@@ -116,6 +117,9 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             }
         }
 
+        var fileTypeCheckList = ImmutableSet.<ExperimentFileType>builder();
+        fileTypeIds.stream().forEach(fileTypeId -> fileTypeCheckList.add(ExperimentFileType.fromId(fileTypeId)));
+
```
Comment (haideriqbal) [inline src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java]: I guess formatting code according to following will increase the readability of code.
`fileTypeIds`
           `.stream()`
           `.foreach(fileTypeId ->`
                   `fileTypeChecklist.add(...)`

File: src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java [hunk 2]
```diff
@@ -165,15 +175,17 @@ public String validateExperimentsFiles(
             }
         });
 
-        var fileTypeCheckList = ImmutableList.of(
-                ExperimentFileType.QUANTIFICATION_RAW,
-                ExperimentFileType.NORMALISED,
-                ExperimentFileType.EXPERIMENT_DESIGN);
+        var fileTypeCheckList = ImmutableSet.<ExperimentFileType>builder();
+        fileTypeIds.stream().forEach(fileTypeId -> fileTypeCheckList.add(ExperimentFileType.fromId(fileTypeId)));
```
Comment (haideriqbal) [inline src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java]: Same comment as above.

File: src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java [hunk 3]
```diff
@@ -165,15 +175,17 @@ public String validateExperimentsFiles(
...
+//        var fileTypeCheckList = ImmutableList.of(
+//                ExperimentFileType.QUANTIFICATION_RAW,
+//                ExperimentFileType.NORMALISED,
+//                ExperimentFileType.EXPERIMENT_DESIGN);
```
Comment (haideriqbal) [inline src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java]: Please remove the commented out code.

Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java]: Removed.

File: src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java (hunk 2 above)
Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java]: Refactored.

File: src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java (hunk 1 above)
Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java]: Refactored.

File: src/test/java/uk/ac/ebi/atlas/download/FileDownloadControllerTest.java [hunk 4]
```diff
@@ -122,12 +123,27 @@ public void testBadAccessions() {
     @Test
     public void testEmptyAccessions() {
```
Comment (alfonsomunozpomer) [inline src/test/java/uk/ac/ebi/atlas/download/FileDownloadControllerTest.java]: Be judicious (but mostly consistent) about the use of blank lines. Have a look at your code overall to remove any blank line that doesn’t improve code readability.

**Pull request merged**